File: ios/CodePush/CodePushModule.swift
import Foundation

@objc(CodePushModule)
class CodePushModule: NSObject {
  private let downloadManager = DownloadManager.shared
  private let codePushManager = CodePushManager.shared
  private let diskManager = DiskManager.shared

  @objc func initialize(_ host: String) {
    #if DEBUG
      return
    #endif
    
    downloadManager.downloadBundle(host: host, currentVersion: codePushManager.currentVersion) { error, path in
      if let path {
        try? DiskManager.shared.unpack(archive: path)
      }
      
      if let error {
        // Handle error
      }
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

File: android/app/src/main/java/com/codepushexample/codepush/module/CodePushModule.kt
package com.codepushexample.codepush.module

import com.codepushexample.codepush.CodePushManager
import com.codepushexample.codepush.DiskManager
import com.codepushexample.codepush.CodePushDownloadManager
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlin.coroutines.CoroutineContext

class CodePushModule(
    context: ReactApplicationContext,
    val diskManager: DiskManager,
    val codePushManager: CodePushManager,
    val downloadManager: CodePushDownloadManager
) : ReactContextBaseJavaModule(context), CoroutineScope {
    override fun getName(): String = "CodePushModule"

    @ReactMethod
    fun initialize(host: String) {
        downloadManager.setup(host)

        launch {
            try {
                val file = downloadManager.loadInBackground() ?: return@launch
                diskManager.unpack(file)
            } catch (ex: Throwable) {
                val x = 0;
                // Handle error
            }
        }
    }

    override val coroutineContext: CoroutineContext
        get() = Dispatchers.IO
}